File: benchmarks/list_sum.go
package main

import (
	"fmt"
	"time"
)

func main() {
	t0 := time.Now()
	acc := []int{}
	i := 0
	for i < 3000000 {
		acc = append(acc, i)
		i = i + 1
	}
	total := 0
	for _, x := range acc {
		total = total + x
	}
	fmt.Println(total)
	fmt.Printf("bench_ns=%d\n", time.Since(t0).Nanoseconds())
}

File: benchmarks/list_sum.rs
use std::time::Instant;

fn main() {
    let t0 = Instant::now();
    let mut acc = Vec::new();
    let mut i = 0;
    while i < 3000000 {
        acc.push(i);
        i += 1;
    }
    let mut total = 0;
    for x in acc {
        total += x;
    }
    println!("{}", total);
    println!("bench_ns={}", t0.elapsed().as_nanos());
}
